cmd/bootkube/render.go

```diff
@@ -196,3 +217,28 @@ func altNamesFromURLs(urls []*url.URL) *tlsutil.AltNames {
 	}
 	return &an
 }
+
+func pickDNSServiceIP(serviceCIDR string) (string, error) {
+	ip, ipnet, err := net.ParseCIDR(serviceCIDR)
+	if err != nil {
+		return "", err
+	}
+	// pick the 10th IP as a conventional default
+	for i := 0; i < 10; i++ {
+		incIPv4(ip)
```

Comment:
For users who use the `render` subcommand, it MUST correspond to the on-host kubelet's flag when self-hosted Kublet isn't in use. It is valuable to have it be deterministically calculated for users who prefer to use a particular service range.

Perhaps render outputs a message about this. On master, the hack examples happen to correspond to the DNS service IP hard-coded in the bootkube binary and rendered to manifests.